src/GovUk.Education.ExploreEducationStatistics.Admin/Services/Public.Data/DataSetService.cs

```diff
@@ -60,6 +62,29 @@ public class DataSetService(
             });
     }
 
+    public async Task<Either<ActionResult, DataSetVersionSummaryViewModel>> GetVersion(
+        Guid releaseVersionId,
+        Guid dataSetId,
+        string dataSetVersion,
+        CancellationToken cancellationToken = default)
+    {
+        return await CheckReleaseVersionExists(releaseVersionId, cancellationToken)
+            .OnSuccess(userService.CheckCanViewReleaseVersion)
+            .OnSuccessCombineWith(async _ => await CheckVersionExists(dataSetId, dataSetVersion))
+            .OnSuccessDo(async combinedReleaseAndDataSetVersion =>
+            {
+                var (releaseVersion, dataSetVersion) = combinedReleaseAndDataSetVersion;
+
+                return await CheckDataSetVersionBelongsToReleaseVersion(dataSetVersion, releaseVersion);
+            })
+            .OnSuccess(combinedReleaseAndDataSetVersion =>
+            {
+                var (releaseVersion, dataSetVersion) = combinedReleaseAndDataSetVersion;
+
+                return MapDataSetVersion(dataSetVersion, releaseVersion);
```

Comment:
Maaaaaaaybe...

In which case, this endpoint wouldn't even need to be passed a `releaseVersionId` as it is currently. It could just accept the `dataSetVersionId` and work from there?

I'm not sure what the answer is here either

Reply:
Yeah that's what I was thinking too, but I'm not sure of the answer.